test/Factory.js

```diff
@@ -0,0 +1,60 @@
+﻿"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "require", "./amchartFactory", "./c3chartFactory", "./chartFactory", "./commonFactory", "./formFactory", "./googleFactory", "./graphFactory", "./layoutFactory", "./mapFactory", "./marshallerFactory", "./otherFactory", "./treeFactory"], factory);
+    } else {
+        root.test_Factory = factory(root.d3, root.require, root.test_amchartFactory, root.test_c3chartFactory, root.test_chartFactory, root.test_commonFactory, root.test_formFactory, root.test_googleFactory, root.test_graphFactory, root.test_layoutFactory, root.test_mapFactory, root.test_marshallerFactory, root.test_otherFactory, root.test_treeFactory);
+    }
+}(this, function (d3, require, amchartFactory, c3chartFactory, chartFactory, commonFactory, formFactory, googleFactory, graphFactory, layoutFactory, mapFactory, marshallerFactory, otherFactory, treeFactory) {
+    var bundles = { common: commonFactory, chart: chartFactory, amchart: amchartFactory, google: googleFactory, c3chart: c3chartFactory, map: mapFactory, tree: treeFactory, graph: graphFactory, other: otherFactory, form: formFactory, layout: layoutFactory, marshaller: marshallerFactory };
+    var bundlesCatMap = {};
+    var bundlesWidgetMap = {};
+    d3.map(bundles).entries().forEach(function (bundle) {
+        var folderID = bundle.key;
+        d3.map(bundle.value).entries().forEach(function (widget) {
+            var widgetID = widget.key;
+            d3.map(widget.value).entries().forEach(function (sample) {
+                var sampleID = sample.key;
+                var widgetPath = "src/" + folderID + "/" + widgetID;
+                var widgetTestData = {
+                    folder: folderID,
+                    file: widgetID,
+                    sample: sampleID,
+                    widgetPath: widgetPath,
```

Comment:
I think the following variables are badly named:
```
var bundlesCatMap
var bundlesWidgetMap
```
And should really be called:
```
var folderName;
var fileName;
```